### PR TITLE
Feature/wa 01 app launcher

### PR DESCRIPTION
**Pull request by edgarquispe**
app launcher and feature selecction e.g. Accounts, Groups, Opportunities, etc

### CODE REVIEW

File: src/test/resources/features/Signin.feature [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Feature: Signin
 
   Scenario: Salesforce sign in
-    Given I sign in as "username" user and "password"
+    Given I sign in as "edgariverq@personal.com" user and "EEE199909049qqq" as password
```
Comment (acamacho82) [inline src/test/resources/features/Signin.feature:4]: remove your user credentials

File: src/test/resources/features/account.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Feature: Projects
```
Comment (acamacho82) [inline src/test/resources/features/account.feature]: This should be Accounts feature

File: src/test/resources/features/account.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  Scenario: Create Basic Account
+    Given I sign in as "account" user and "Password" as password
+    And I click on the App Launcher button and select item "Opportunities"
+#    When I click on New button
+#    And I set the account form
+#      | AccountName    | BasicAcc01 |
+#    Then I validate the account name is "BasicAcc01"
```
Comment (acamacho82) [inline src/test/resources/features/account.feature]: Remove unnecessary comments

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+public class AppLaunchPage extends AbstractPage {
+    private static final String PRIVACY_CSS = "//a[@title='%s']";
+    @FindBy(css = ".slds-icon-waffle_container")
+    private WebElement AppLauncherButton;
+
+    @FindBy(css = "#password")
+    private WebElement passwordTextField;
+
+    @FindBy(css = "#Login")
+    private WebElement loginButton;
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: If I'm not wrong there aren't #Login and #password locators in the AppLaunch page

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    public Header LaunchApp(String feature) {
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: As far as I can understand, this method click items and not apps, so I think this should be called `clickOnItem(String item)` or `launchItem(String item)`

Why this method returns a Header?
it is expected to use the Header page object all the time once an item is clicked?
I think this should be a `void` method instead

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.jalasoft.sdfc.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class AppLaunchPage extends AbstractPage {
+    private static final String PRIVACY_CSS = "//a[@title='%s']";
+    @FindBy(css = ".slds-icon-waffle_container")
+    private WebElement AppLauncherButton;
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: I think this should be in the Header and once it  is clicked the method should return an AppLaunchPage object

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package com.jalasoft.sdfc.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class AppLaunchPage extends AbstractPage {
+    private static final String PRIVACY_CSS = "//a[@title='%s']";
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: I think it should be called ITEM_LINK, ITEM_NAME or something like that

**Review by acamacho82 — CHANGES_REQUESTED** [review]
Not bad, just need some changes. Let me know if you have questions

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 7 above)
Comment (bruxita2312) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: Could be "SETUP_ITEM_LINK_CSS" instead PRIVACY_CSS

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 4 above)
Comment (bruxita2312) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: Yes, those items should be removed from AppLaunch page.

**Review by bruxita2312 — CHANGES_REQUESTED** [review]

File: src/test/resources/features/Signin.feature (hunk 1 above)
Comment (edgarquispe) [reply src/test/resources/features/Signin.feature:4]: please review the last commit

File: src/test/resources/features/account.feature (hunk 2 above)
Comment (edgarquispe) [reply src/test/resources/features/account.feature]: please review the last commit

File: src/test/resources/features/account.feature (hunk 3 above)
Comment (edgarquispe) [reply src/test/resources/features/account.feature]: please review the last commit

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 4 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: please review the last commit

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 5 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: please review the last commit

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 6 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: please review the last commit

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 7 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java]: please review the last commit

File: src/main/java/com/jalasoft/sdfc/pages/Home.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.jalasoft.sdfc.pages;
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/Home.java]: I do not think that this file is part of the scope of this task, and besides, the WebElement and the method are the same as those found in the Header file.

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+public class AppLaunchPage extends AbstractPage {
+
+    private static final String ITEM_LINK = "//a[@title='%s']";
+
+
+    public void clickOnItem(String item) {
+
+        action.click(By.xpath(String.format(ITEM_LINK, item)));
+
```
Comment (acamacho82) [inline src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java:12]: remove the blank lines that are around the action.click ()

**Review by acamacho82 — CHANGES_REQUESTED** [review]
last changes, just remove the Home.java file since it is not part of the scope of this task, and the blank lines  in the clickOnItem() method

File: src/main/java/com/jalasoft/sdfc/pages/Home.java (hunk 8 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/Home.java]: deleted, please review last commit.

File: src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java (hunk 9 above)
Comment (edgarquispe) [reply src/main/java/com/jalasoft/sdfc/pages/AppLaunchPage.java:12]: removed, please review last commit.

**Review by acamacho82 — APPROVED** [review]

**Review by mfuentes-ME — APPROVED** [review]

**Review by bruxita2312 — APPROVED** [review]

**Pull request merged**